pkg/services/alerting/notifiers/slack.go

```diff
@@ -133,14 +156,69 @@ func (this *SlackNotifier) Notify(evalContext *alerting.EvalContext) error {
 	if this.Recipient != "" {
 		body["channel"] = this.Recipient
 	}
-
 	data, _ := json.Marshal(&body)
 	cmd := &m.SendWebhookSync{Url: this.Url, Body: string(data)}
-
 	if err := bus.DispatchCtx(evalContext.Ctx, cmd); err != nil {
 		this.log.Error("Failed to send slack notification", "error", err, "webhook", this.Name)
 		return err
 	}
+	if this.Token != "" {
+		slackUploadUrl := "https://slack.com/api/files.upload"
+		if evalContext.ImageOnDiskPath == "" {
+			evalContext.ImageOnDiskPath = "public/img/mixed_styles.png"
```

Comment:
If ImageOnDiskPath is empty we should just ignore uploading it.

Reply:
I wanted to make sure a "test" notification would perform as expected and actually upload an image, otherwise I'm not sure how you'd confirm the token you've provided works with that API endpoint.

Reply:
That makes sense. But if people disable images for alert notifications this image will still be uploaded.

Reply:
Ah, you are correct. That should be addressed in https://github.com/grafana/grafana/pull/9392/commits/be0d47146751733e60807240c5082ea777a74109.